pkg/cgroup/resolve_container.go

```diff
@@ -231,49 +222,23 @@ func validContainerID(id string) string {
 
 // Get containerID from path. cgroup v1 and cgroup v2 will use different regex
 func extractPodContainerIDfromPath(path string) (string, error) {
-	cgroup := config.GetCGroupVersion()
-	return extractPodContainerIDfromPathWithCgroup(path, cgroup)
+	return extractPodContainerIDfromPathWithCgroup(path)
 }
 
-func extractPodContainerIDfromPathWithCgroup(path string, cgroup int) (string, error) {
+func extractPodContainerIDfromPathWithCgroup(path string) (string, error) {
 	if path == unknownPath {
 		return utils.SystemProcessName, fmt.Errorf("failed to find pod's container id")
 	}
-	// as the container ID is located at the end of the path, we remove the beginning of the string to prevent bugs, as seen in issue #923
-	split := strings.Split(path, "/")
-	size := len(split)
-	path = split[size-1 : size][0]
-
-	if regexFindContainerIDPath.MatchString(path) {
-		sub := regexFindContainerIDPath.FindAllString(path, -1)
-		for _, element := range sub {
-			if cgroup == 2 && (strings.Contains(element, "-conmon-") || strings.Contains(element, ".service")) {
-				return "", fmt.Errorf("process is not in a kubernetes pod")
-				// TODO: we need to extend this to include other runtimes
-			} else if strings.Contains(element, "crio") || strings.Contains(element, "docker") || strings.Contains(element, "containerd") {
-				containerID := regexReplaceContainerIDPathPrefix.ReplaceAllString(element, "")
-				containerID = regexReplaceContainerIDPathSufix.ReplaceAllString(containerID, "")
-				return validContainerID(containerID), nil
-			}
-		}
-	}
-	// TODO: need to get a path to verify this case
-	// as some platforms (e.g. RHEL) have a different cgroup path, if the cgroup path has information from a pod and we can't get the container id
-	// with the previous regex, we'll try to get it using a different approach
-	if regexFindContainerIDPath2.MatchString(path) {
-		sub := regexFindContainerIDPath2.FindAllString(path, -1)
-		for _, containerID := range sub {
-			return validContainerID(containerID), nil
-		}
-	}
-	// some systems, such as minikube, create a different path that has only the kubepods keyword
-	if strings.Contains(path, "kubepods") {
-		tmp := strings.Split(path, "/")
-		containerID := tmp[len(tmp)-1]
-		return validContainerID(containerID), nil
-	}
 
-	return utils.SystemProcessName, fmt.Errorf("failed to find pod's container id")
+	path = strings.TrimSuffix(path, "/container")
+	path = strings.TrimSuffix(path, ".scope")
+
+	// get the last 64 characters of the path
```

Comment:
can you post the reference to this 64-char path length?

Reply:
https://docs.docker.com/engine/reference/run/#container-identification

I've tried looking for a reference in the OCI spec but I couldn't find one, but from what I can see this appears to be the de-facto standard.

We could split the path string on all `/` and use the last path component, which wouldn't cause an issue in the case of something that's not exactly 64 chars long.

Reply:
I could yes, but consider this - it's a 10x performance improvement over the legacy code + all of our unit tests (including the newly added ones from podman) pass. Not only that, it's called on when the K8s watcher isn't enabled.

However, when the k8s watcher isn't enabled (i.e linux only, no k8s) this function is in the hot path - i.e it's called on in `createContainerStatsIfNotExist` which is called multiple times every time we collect metrics. Adding a fallback path is expensive in terms of performance - I'd rather that users submit issues so we can expand our tests cases and have a performant and correct solution. 

```
go test -benchmem -run=^$ -bench ^BenchmarkExtractPodContainerIDfromPathWithCgroup github.com/sustainable-computing-io/ke
pler/pkg/cgroup 
goos: linux
goarch: amd64
pkg: github.com/sustainable-computing-io/kepler/pkg/cgroup
cpu: AMD Ryzen 5 5600G with Radeon Graphics         
BenchmarkExtractPodContainerIDfromPathWithCgroup-12              1222920               930.0 ns/op             0 B/op          0 allocs/op
BenchmarkExtractPodContainerIDfromPathWithCgroupLegacy-12         123741              9918 ns/op             595 B/op          9 allocs/op
PASS
ok      github.com/sustainable-computing-io/kepler/pkg/cgroup   3.463s
```